venv/lib/python3.12/site-packages/anyio/_core/_asyncio_selector_thread.py

```
@@ -0,0 +1,167 @@
+
+    def add_reader(self, fd: FileDescriptorLike, callback: Callable[[], Any]) -> None:
+        loop = asyncio.get_running_loop()
+        try:
+            key = self._selector.get_key(fd)
+        except KeyError:
+            self._selector.register(fd, EVENT_READ, {EVENT_READ: (loop, callback)})
+        else:
+            if EVENT_READ in key.data:
+                raise ValueError(
+                    "this file descriptor is already registered for reading"
+                )
```

Ruff (0.13.3)

Avoid specifying long messages outside the exception class

(TRY003)